docs/introduction.md

```diff
@@ -0,0 +1,282 @@
+During unification, a denormalized substitution is considered an error
+because we have failed to produce a substitution which unifies the given patterns.
+During predicate simplification, we are more flexible;
+it is entirely reasonable to generate conditions such as `x = x + y` in this context.
+
+References:
+
+- `Kore.Unification.SubstitutionNormalization`
+- `Kore.Step.Simplification.SubstitutionSimplifier`
+
+### Matching
+
+An ad hoc subset of patterns can be matched, based on the usual needs of the frontend and semantics developers.
+Matching is implemented sequentially, so there is no separate substitution normalization step.
+
+References:
+
+- `Kore.Step.Axiom.Matcher`
+
+### Semantic rules
+
+Semantic rules are applied by the following procedure:
+
+1. Unify (as described above) the left-hand side of the rule with the current configuration's term.
+1. Conjoin the `requires` clause with the unification conditions and simplify.
+1. Conjoin with the initial conditions (from the current configuration) and simplify.
+1. Check that the conditions include substitutions for every variable on the left-hand side of the rule.
+1. Conjoin with the `ensures` clause.
+1. Instantiate the rule with the combined conditions.
+1. Calculate the _remainder_ from the (unification+requires) conditions of all applied rules.
+
+The procedures to apply one-path and all-path reachability claims is described in other documents.
+
+References:
+
+- `Kore.Step.Step`
+
+### Function rules
+
+Function rules are instantiated by matching, as described above.
+Function evaluation occurs during simplification, but not during unification.
+Function evaluation _may_ split the configuration
+if multiple branches of a function's definition match the configuration;
+this point is planned to change.
+
+References:
+
+- `Kore.Step.Function.Evaluator`
+- `Kore.Step.Axiom.EvaluationStrategy`
+
+### Simplification
+
+Patterns are simplified from the bottom up.
+The top-level condition of the configuration is passed down as context for function and predicate evaluation.
+New conditions are accumulated as simplification moves up the syntax tree,
+requiring substitution normalization at the top.
+`TermLike` is simplified in a single pass, but this is planned to change.
+`Condition` is simplified in a loop where new substitutions are applied and the result is re-simplified.
+
+References:
+
+- `Kore.Step.Simplification.TermLike`
+- `Kore.Step.Simplification.Condition`
+
+### Refuting predicates
+
+Some predicates can be evaluated by an external solver;
+in theory this can be any solver with an SMT-LIB 2 interface, but in practice we use Z3.
+Generally, we only check satisfiability of a predicate,
+using the solver to refute impossible execution paths.
+Predicate logic connectives are translated to their corresponding connectives in the solver.
+`\equals` can be translated if its arguments are in `Int` or `Bool`.
```

Comment:
Equals also works for constructor-based stuff, here's the call stack:
https://github.com/kframework/kore/blob/master/kore/src/Kore/Step/SMT/Translate.hs#L103
https://github.com/kframework/kore/blob/master/kore/src/Kore/Step/SMT/Translate.hs#L152
https://github.com/kframework/kore/blob/master/kore/src/Kore/Step/SMT/Translate.hs#L259
https://github.com/kframework/kore/blob/master/kore/src/Kore/Step/SMT/Translate.hs#L233
https://github.com/kframework/kore/blob/master/kore/src/Kore/Step/SMT/Resolvers.hs#L51